lang/strings.ts

```diff
@@ -0,0 +1,81 @@
+import { EOL } from 'https://deno.land/std@0.145.0/fs/eol.ts';
+import { isNil } from './values.ts';
+
+/**
+ * Removes a trailing new line (either LF or CRLF) from the string if there is one.
+ *
+ * @param text The text to be altered
+ * @returns The altered text
+ */
+export function trimTrailingNewline(text: string): string {
+  if (text.endsWith(EOL.CRLF)) {
+    return stripSuffix(text, EOL.CRLF);
+  } else {
+    return stripSuffix(text, EOL.LF);
+  }
+}
+
+/**
+ * Removes the specified prefix from the string if it exists.
+ *
+ * @param text The text to be altered
+ * @param prefix The prefix to remove from the text
+ * @returns The altered text
+ */
+export function stripPrefix(text: string, prefix: string): string {
+  return text.startsWith(prefix) ? text.substring(prefix.length) : text;
+}
+
+/**
+ * Removes the specified suffix from the string if it exists.
+ *
+ * @param text The text to be altered
+ * @param suffix The suffix to remove from the text
+ * @returns The altered text
+ */
+export function stripSuffix(text: string, suffix: string): string {
+  return text.endsWith(suffix) ? text.substring(0, text.length - suffix.length) : text;
+}
+
+/**
+ * Formats the template string with the leading/trailing newlines and largest common line indentations removed.
+ *
+ * This is useful for multiline template strings.
+ *
+ * @example
+ * const generatedCode = trimAndDedent`
+ *   service ${serviceName} {
+ *     rpc Delete${subjectName}(Delete${subjectName}Request) returns (Delete${subjectName}Response);
+ *     rpc Create${subjectName}(Create${subjectName}Request) returns (Create${subjectName}Response);
+ *   }
+ * `;
+ */
```

Comment:
Nit: no jsdocs for the function (but I am loving these examples)-- if it's meant as a tag template function, that should be called out explicitly.

Reply:
Good catch, thanks!